components/kyma-environment-broker/common/orchestration/strategies/parallel.go

```diff
@@ -16,42 +17,43 @@ type Executor interface {
 
 type ParallelOrchestrationStrategy struct {
 	executor Executor
-	log      logrus.FieldLogger
+	dq       workqueue.DelayingInterface
```

Comment:
We should keep `dq` local to execution, otherwise if there were concurrent executions launched by the same strategy ovject, operations would be mixed up in the shared dq. I know right now each orchestration creates a separate strategy object, but the strategy was designed to be thread safe and this change breaks that. 

You could do similar logic which is done with execution specific waitgroup: save it in a map (access protected by the mutex), and when `Cancel(execID)` is called, look up the `dq` based in `execID` and call `Shutdown()` on looked-up object.

Reply:
ok I'm saving now delaying queues into the map and I'm using them in the context of executionID, it should be thread-safe now again